src/components/dashboard/FaceVerification/api/index.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+import axios from 'axios'
+import API from '../../../../lib/API/api'
+import logger from '../../../../lib/logger/pino-logger'
+import goodWallet from '../../../../lib/wallet/GoodWallet'
+import { type FaceVerificationPayload, type FaceVerificationResponse } from './typings'
+
+class FaceVerificationApi {
+  rootApi: typeof API
+
+  logger: any
+
+  lastCancelToken: any = null
+
+  /**
+   * wallet an instance of GoodWallet
+   * @instance {GoodWallet}
+   */
+  wallet: GoodWallet
+
+  constructor(rootApi: typeof API, wallet: GoodWallet, logger: any) {
+    this.rootApi = rootApi
+    this.logger = logger
+    this.wallet = wallet
+  }
+
+  async performFaceVerification(
+    payload: FaceVerificationPayload,
+    progressSubscription?: ({ loaded: number, total: number }) => void
+  ): Promise<FaceVerificationResponse> {
+    let axiosConfig = {}
+    const { rootApi, logger } = this
+
+    const { sessionId } = payload
+
+    this.lastCancelToken = axios.CancelToken.source()
+
+    axiosConfig = {
+      cancelToken: this.lastCancelToken.token,
+      onProgress: progressSubscription,
+    }
+
+    logger.info('performFaceVerification', { sessionId })
+
+    try {
+      const { data: response } = await rootApi.performFaceVerification(
+        payload,
+        this.enrollmentIdentifier(),
+        axiosConfig
+      )
+      const { success, error } = response || {}
+
+      if (!response) {
+        throw new Error('Failed to perform face recognition on server')
+      }
+
+      if (!success) {
+        const exception = new Error(error)
+
+        exception.response = response
+        throw exception
+      }
+
+      logger.info('Face Recognition finished successfull', { response })
+
+      return response
+    } catch (exception) {
+      const { message, response } = exception
+      const { error } = response || {}
+
+      logger.error('Face recognition failed', error || message, exception)
+      throw exception
+    } finally {
+      this.lastCancelToken = null
+    }
+  }
+
+  cancelInFlightRequests() {
+    const { lastCancelToken } = this
+
+    if (!lastCancelToken) {
+      return
+    }
+
+    lastCancelToken.cancel('Face verification timeout')
+    this.lastCancelToken = null
+  }
+
+  enrollmentIdentifier(): string {
+    return this.wallet.getAccountForType('faceVerification').replace('0x', '')
+  }
+
+  async disposeFaceSnapshot(): Promise<void> {
+    const { wallet, rootApi, enrollmentIdentifier } = this
```

Comment:
1. i would remove dependency on wallet from here, and get identifier+signature in method params
2. FYI enrollmentIdentifier is a method and you dont execute it, so .sign actually signs the method and not the identifier

Reply:
1. okay, could we to move enrollmentIdentifier() as `getFaceIdentifier()` to UserStorageClass ? UserStorageClass will use this method on account removal, useFaceVerification hook will import UserStorage and call getFaceIdentifier() to get enrollment id before send enroll ?
2. thanks. missed it up reviewing @StanislavShevchenko 
anyway he would found this issue soon as he works now on 24h enrollment removal ;)